src/mapwindow.jl

```diff
@@ -273,13 +365,100 @@ end
 # This is slightly faster than a circular buffer
 @inline cyclecache(b, x) = b[1], (Base.tail(b)..., x)
 
+
 replace_function(f) = f
-replace_function(::typeof(median!)) = function(v)
-    inds = indices(v,1)
-    Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+
+replace_function(::typeof(median!)) = function(v,m_histogram,mode,window)
+   # Handle the boundary points with mode -1
+    if(mode==-1)
+        inds = indices(v,1)
+        return Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+    else
+        
+        window_size=size(v,1)
+        dims = map(x->x.stop-x.start+1,window)
+        width=window[1].stop-window[1].start+1
+        inds = indices(v,1)
+        if mode == 0
+        # Update the histogram according to new entries
+            for i = first(inds):last(inds)
+                id= trunc(Int64,(v[i]*255))+1
```

Comment:
If the median_histogram method is restricted to N0f8, that won't be problem right?